lib/rake/extensiontask.rb

```diff
@@ -132,10 +133,27 @@ def define_compile_tasks(for_platform = nil, ruby_ver = RUBY_VERSION)
       directory lib_binary_dir_path
       directory stage_binary_dir_path
 
+      directory File.dirname(siteconf_path)
+      # Set paths for "make install" destinations
+      file siteconf_path => File.dirname(siteconf_path) do
+        File.open(siteconf_path, "w") do |siteconf|
+          siteconf.puts "require 'rbconfig'"
+          siteconf.puts "require 'mkmf'"
+          siteconf.puts "dest_path = mkintpath(#{File.expand_path(lib_path).dump})"
```

Comment:
This is the same mechanism as [used in rubygems](https://github.com/rubygems/rubygems/blob/17904c6312bf773e3e3b62589efd4e663f180c21/lib/rubygems/ext/ext_conf_builder.rb#L28-L34). However we're using an absolute `dest_path` here, which has to be passed through `mkintpath` to properly work on Windows. In contrast rubygems uses a relative path and subsequently copies the files from a temporary directory to the destination extension directory.